docs/release-notes/july-2023.md

```diff
@@ -0,0 +1,231 @@
+    ```rust
+    fn f ((x, (y, z)): t) {
+      let x = 2;
+    }
+    ```
+    Tainting the sole argument to this function results in all of the identifiers `x`, `y`, and `z` now being tainted. ([#8216](https://github.com/returntocorp/semgrep/pull/8216))
+* Rust: Added support for ellipsis patterns in attribute argument positions. For example, `#[get(...)]`. ([#8148](https://github.com/returntocorp/semgrep/pull/8234))
+* Rust: Added typed metavariable support for Rust. Users can create `TypedMetavar` using Rust's type annotation syntax `:`. For example, the following rule works for matching `HttpResponseBuilder` type of variables:
+    ```yaml
+    rules:
+    - id: no-direct-response-write
+      patterns:
+      - pattern: '($BUILDER : HttpResponseBuilder).body(...)'
+      - pattern-not: '($BUILDER : HttpResponseBuilder).body("...".to_string())'
+      message: find dangerous codes
+      severity: WARNING
+      languages: [rust]
+    ```
+* Rust: Added the ability to taint macro calls through its arguments, in macro calls with multiple arguments. (#[8209](https://github.com/returntocorp/semgrep/pull/8209))
+* Matching: Added the ability to use metavariables in parameters to match more sophisticated kinds of parameters. In particular, metavariables should now be able to match `self` parameters, such as in Rust. For example: 
+    ```rust
+    fn $F($X, ...) { ... }
+    ```
+    should match:
+    ```rust
+    fn $F(self) { }
+    ```
+* Added support for naming propagation when the left-hand side (LHS) of a variable definition is an **identifier pattern**. In certain languages such as Rust, the variable definition is parsed as a pattern assignment, for example:
+    ```rust
+    let x: SomeType = SomeFunction();
+    ```
+   This commit ensures that the annotated type is propagated to the identifier pattern on the left-hand side (LHS) of the assignment, thus ensuring proper naming behavior. 
+* **Taint-mode:** Added experimental `control: true` option to `pattern-sources`. For example:
+    ```yaml
+    pattern-sources:
+      - control: true
+          pattern: source(...)
+    ```
+    * Such sources taint the "control flow" (or the program counter) so that it is possible to implement reachability queries that do not require the flow of any data. Thus, Semgrep reports a finding in the code below, because after source() the flow of control will reach sink(), even if no data is flowing between both:
+    ```yaml
+    def test():
+      source()
+      foo()
+      bar()
+      #ruleid: test
+      sink()
+    ``` (pa-2958)
+* Taint mode: Taint sanitizers will be included in matching explanations. ([#8383](https://github.com/returntocorp/semgrep/issues/8383))
+
+### Fixed
+
+- Semgrep no longer crashes when running `--test`. (#[8360](https://github.com/returntocorp/semgrep/pull/8360))
```

Comment:
Unfortunately this isn't the case - that issue is persisting. We should remove this line.